main.js

```diff
@@ -273,7 +304,8 @@ loops = {
 			miner.value = parseInt($(`#${miner.name}-amount`).text());
 			// this is available to buy
 			if ($(`#${miner.name}`).attr("style") === "opacity: 1;") {
-				if (miner.value < config.maxMinerLevel) {
+				// buy more quantum servers and botnets, buy botnets at the same rate as the quantum servers.
+				if (miner.value < config.maxMinerLevel || (miner.name === "shop-quantum-server" || miner.name === "shop-bot-net" && miner.value < vars.minerStatus[5].value) && miner.value < config.maxQBLevel) {
```

Comment:
That's a pretty big `if`.  Do you think you could restructure to have fewer conditions?  Perhaps trap the inverse?  Instead of when to click, try escaping the `if` early by trapping on when NOT to click?  This is just off the top of my head, so that may not work, but see if it can be condensed a bit.